util/cf_options.cc

```diff
@@ -75,18 +75,14 @@ ImmutableCFOptions::ImmutableCFOptions(const ImmutableDBOptions& db_options,
       max_subcompactions(db_options.max_subcompactions) {}
 
 // Multiple two operands. If they overflow, return op1.
-uint64_t MultiplyCheckOverflow(uint64_t op1, int op2) {
-  if (op1 == 0) {
+uint64_t MultiplyCheckOverflow(uint64_t op1, double op2) {
+  if (op1 == 0 || op2 <= 0) {
     return 0;
   }
-  if (op2 <= 0) {
+  if (std::numeric_limits<uint64_t>::max() / op1 < op2) {
     return op1;
```

Comment:
And multiplying by zero is not an overflow and I don't see why it made sense to return op1 here. This function is really weird to me, perhaps I'm missing something.